db_checker.py

```diff
@@ -37,122 +39,128 @@
     'cdeg/ss',  # Needed by the GORC. Latter is a special case because cdeg/s^2 too long}
 }
 
-dbs = list()
-
 
-def err_src_fmt(db, rec):
+def ignore(dbs, message="Skipped"):
     """
-    Create an error string for the db record and file name
-    Args:
-        db: db record that error was in
-        rec: the record the error was in
-
-    Returns: location of the error
-
+    Decorator to skip tests on certain DBs or paths
+    :param dbs: DBs or paths to skip on.
+    :param message: skip message
     """
-    return str(rec) + " in " + str(db)
+    def decorator(func):
+        def wrapper(self, *args, **kwargs):
+            if any(db in self.db.directory for db in dbs):
+                self.skipTest(message)
+            func(self, *args, **kwargs)
+        return wrapper
+    return decorator
 
 
 class TestPVUnits(unittest.TestCase):
     """
     Test class for db records
     """
 
+    def __init__(self, methodName, db=None):
+        super(TestPVUnits, self).__init__(methodName=methodName)
+        self.db = db
+
+    @ignore(["superlogics.db", "lakeshore336.db", "motor.db"])
+    @ignore(["EPICS_V4"], "Vendor-supplied DBs")
+    @ignore(["DbUnitChecker"], "DB unit checker contains tests that should fail.")
     def test_multiple_pvs_warning(self):
         """
         This method warns if there are multiple PVs with the same name in the project
         """
-        for db in dbs:
-            dups = defaultdict(list)  # Makes a dict of lists
-            for rec in db.records:
-                dups[str(rec.pv)].append(rec)
+        err = 0
+        failure_message = "Multiple fields on PVs in {}\n".format(self.db.directory)
+
+        dups = defaultdict(list)  # Makes a dict of lists
+        for rec in self.db.records:
+            dups[str(rec.pv)].append(rec)
 
-            for k, v in dups.iteritems():
-                if len(v) > 1:
-                    print "WARNING: Multiple instances of " + err_src_fmt(db, k)
+        for k, v in dups.iteritems():
```

Comment:
Should we use `items()` here? My understanding is that it's more memory intensive in Python 2 because it returns a copied object rather than an iterator. However, `iteritems` isn't available in Python 3 and `items` does what `iteritems` does now. I think I'd pay the upfront memory cost for better compatibility. One less thing to do when we migrate.